lib/arel/middleware/database_executor.rb

```diff
@@ -46,7 +46,16 @@ def call(next_arel)
 
       def execute_sql(next_arel)
         sql, binds = next_arel.to_sql_and_binds
+
+        context[:cache].set(
```

Comment:
I guess the transformations done on this query should be part of the cache key 🤔

Reply:
On a different note, I'm not too excited about passing lots of responsibility and internal transformation details to the caching implementation. I'm considering adding a bit more logic to the chain (this module) to make it more opinionated about what to cache.